cola/widgets/branch.py

```diff
@@ -0,0 +1,520 @@
+                                                             info, ok_btn):
+            remote = False
+            root = self.tree_helper.get_root(self.selected_item())
+            if NAME_REMOTE_BRANCH == root.name:
+                remote = True
+
+            if remote is False:
+                self.git_action_async('delete_local', [branch])
+            else:
+                rgx = re.compile(r'^(?P<remote>[^/]+)/(?P<branch>.+)$')
+                match = rgx.match(branch)
+                if match:
+                    remote = match.group('remote')
+                    branch_name = match.group('branch')
+                    self.git_action_async('delete_remote',
+                                          [remote, branch_name],
+                                          update_remotes=True)
+
+    def merge_action(self):
+        branch = self.tree_helper.get_full_name(
+            self.selected_item(),
+            SEPARATOR_CHAR)
+
+        if branch != self.current_branch:
+            self.git_action_async('merge', [branch], refresh_tree=True)
+
+    def checkout_action(self):
+        branch = self.tree_helper.get_full_name(self.selected_item(),
+                                                SEPARATOR_CHAR)
+
+        if branch != self.current_branch:
+            self.git_action_async('checkout', [branch])
+
+
+class BranchTreeWidgetItem(QtWidgets.QTreeWidgetItem):
+    def __init__(self, name, icon):
+        QtWidgets.QTreeWidgetItem.__init__(self)
+        self.name = name
+
+        self.setIcon(0, icon)
+        self.setText(0, name)
+        self.setToolTip(0, name)
+        self.setFlags(Qt.ItemIsEnabled | Qt.ItemIsSelectable)
+
+    # fix key == Qt.Key_Left and index.parent().isValid() error throw
+    # in standard.py when navigating with keyboard and press left key
+    @staticmethod
+    def rowCount():
+        return 1
+
+
+class BranchesTreeHelper(object):
+    @staticmethod
+    def group_branches(list_branches, separator_char):
+        """Convert a list of delimited strings to a nested tree dict"""
+        result = {}
+        for item in list_branches:
+            tree = result
+            for part in item.split(separator_char):
+                tree = tree.setdefault(part, {})
+
+        return result
+
+    @staticmethod
+    def create_top_level_item(name, dict_children, child_icon):
+        """Create a top level tree item and its children """
+
+        def create_children(grouped_branches):
+            """Create children items for a tree item"""
+            result = []
+            for key in grouped_branches.keys():
+                item = BranchTreeWidgetItem(key, child_icon)
+                item.addChildren(create_children(grouped_branches[key]))
+
+                if item.childCount() > 0:
+                    item.setIcon(0, icons.ellipsis())
+
+                result.append(item)
+
+            result.sort(key=lambda x: x)
```

Comment:
The `key=lambda x: x` might not do anything.. maybe the `key=...` argument can be removed?